src/storage-client/src/controller.rs

```diff
@@ -300,7 +300,10 @@ pub trait StorageController: Debug {
 
     /// Allow this controller and instances controlled by it to write to
     /// external systems.
-    async fn allow_writes(&mut self);
+    ///
+    /// Caller must provide a `register_ts`, the timestamp at which any tables
+    /// that are known to the controller should be registered in the txn system.
+    async fn allow_writes(&mut self, register_ts: Option<Self::Timestamp>);
```

Comment:
I think the docstrings (here and for `Controller::allow_writes`) should explain why this is an `Option` and that we will panic if you provide `None` but also want to initialize tables.